Sources/tart/VMDirectory.swift

```diff
@@ -161,6 +181,175 @@ struct VMDirectory: Prunable {
     try diskFileHandle.close()
   }
 
+  private func resizeASIFDisk(_ sizeGB: UInt16) throws {
+    guard let diskutilURL = resolveBinaryPath("diskutil") else {
+      throw RuntimeError.FailedToResizeDisk("diskutil not found in PATH")
+    }
+
+    // First, get current disk image info to check current size
+    let infoProcess = Process()
+    infoProcess.executableURL = diskutilURL
+    infoProcess.arguments = ["image", "info", "--plist", diskURL.path]
+
+    let infoPipe = Pipe()
+    infoProcess.standardOutput = infoPipe
+    infoProcess.standardError = infoPipe
+
+    do {
+      try infoProcess.run()
+      infoProcess.waitUntilExit()
+
+      let infoData = infoPipe.fileHandleForReading.readDataToEndOfFile()
+
+      if infoProcess.terminationStatus != 0 {
+        let output = String(data: infoData, encoding: .utf8) ?? "Unknown error"
+        throw RuntimeError.FailedToResizeDisk("Failed to get ASIF disk info: \(output)")
+      }
+
+      // Parse the plist to get current size
+      do {
+        let plist = try PropertyListSerialization.propertyList(from: infoData, options: [], format: nil) as? [String: Any]
+
+        if let plist = plist {
+          // Look for size information in the "Size Info" dictionary
+          var currentSizeBytes: UInt64?
+
+          // For ASIF images, the size is in the "Size Info" dictionary under "Total Bytes"
+          if let sizeInfo = plist["Size Info"] as? [String: Any],
```

Comment:
This could can be simplified and made more type-safe by using [`PropertyListDecoder`](https://developer.apple.com/documentation/foundation/propertylistdecoder) instead.